src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private static final int ADDITIONAL_SIZE = 5;
+    private static final int NOT_FOUND = -1;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeArray();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
```

Comment:
extract to the method
```suggestion
        checkIndexForAdd(index);
```